### ISSUE TITLE
[SMC] rflash option needed to update PNOR or BMC individually for P9 OpenPower servers.

### ISSUE THREAD

**Issue by djmccoyibm** [xcat2/xcat-core#5499]
rflash for a P9 openpower system today expects the target directory -d to contain the images that need to be updated across the nodelist or group provided.  However, the openpower specification for 9006 or 5104 series servers requests that both BMC and PNOR files along with the pUpdate to be used to perform the update be placed in one directory.  In xcat clusters, a group of 256 or more servers being updated may require that only PNOR or BMC be updated seperately.  If rflash blindly updates both, the network traffic can be doubled, and also be run nearly concurrently resulting in serious network overload.  One way to reduce the network traffic (~1/2) is to allow an option to only update PNOR or BMC.  Today, to accomplish this, the client has to remove the file from the directory that should not be updated.  While moving the file around is fairly simple on base servers, in a clustered filesystem, change control may make this significantly more difficult, while an option in rflash could easily be used.

**Comment by whowutwut** [comment]
@bybai @xuweibj Is it possible to quickly create a PR to support this?

**Comment by bybai** [comment]
Hi @djmccoyibm and @whowutwut ,
Just as @xuweibj mentioned:
1,  To activate and apply BMC:
```
rflash <noderange> -a <BMC_tar_ball>
rpower <noderange> bmcreboot
rpower <noderange> bmcstate
```
2. To activate and apply PNOR:
```
rflash <noderange> -a <PNOR_tar_ball>
rpower <noderange> boot
rflash <noderange> -l
```

**Comment by whowutwut** [comment]
@xuweibj @bybai  Oh, I read the initial post again, I think Doyle did not specifically say which servers this is on.  This issue is on the OpenPower Boston Servers, (SMC) so it's using IPMI.  There's no `-a` option for the supermicro flashing.  @zet809  I think you are in the channel where this is being discussed, please invite @xuweibj and @bybai so they can follow the discussion on Friday.

**Comment by djmccoyibm** [comment]
That is correct.  This is focused on OpenPower Boston servers (MT's 9006 or 5104 for instance).

**Comment by hu-weihua** [comment]
@djmccoyibm, could you help to verify the fix in your environment? If it works, could you close this issue? thank you very much.

**Comment by djmccoyibm** [comment]
@hu-weihua This situation came from a client and I have no xcat environment of this sort to verify with.   @whowutwut How can we complete a verification cycle for this?  Thanks.

**Comment by whowutwut** [comment]
I think a summary is needed here.  The direction taken by @zet809  is to first provide a quick fix of this issue in 2.14.3, via use of ENV variables.  I do think that we should open another issue to track a method to sync up the externals between Supermicro [pUpdate] and OpenBMC [REST] so that the `-a` option is exposed and we have a more consistent external between the two Power9 servers. 

Or we simply leave this open and extend the milestone. 

From #5504, the following has been implemented.  If the firmware is stored here: 
```
[vhu@c910loginx01 xCAT]$ pwd
/xcat/iso/firmware/supermicro/p9/Boston_LC/20180205PNOR_BMC1.16/xCAT
[vhu@c910loginx01 xCAT]$ ls -ltr
total 98560
-rw-r--r-- 1 vhu users 67108992 Jul 17 09:49 p9dsu02052018_IBM_imp_sign.pnor
-rw-r--r-- 1 vhu users 33554432 Jul 17 09:49 SMT_P9_116.bin
-rwxr-xr-x 1 vhu users   133792 Jul 17 09:56 pUpdate
[vhu@c910loginx01 xCAT]$
```

To flash as we currently do today....
`rflash <noderange> -d /xcat/iso/firmware/supermicro/p9/Boston_LC/20180205PNOR_BMC1.16/xCAT/` 

Two new ENV variables are exposed: `XCAT_PUPDATE_BMC,  XCAT_PUPDATE_PNOR`

To flash PNOR only, set to 1 before running `rflash`: 
```
XCAT_PUPDATE_PNOR=1 rflash <noderange> -d /xcat/iso/firmware/supermicro/p9/Boston_LC/20180205PNOR_BMC1.16/xCAT/
```

To flash BMC only, set to 1 before running `rflash`: 
```
XCAT_PUPDATE_BMC=1 rflash <noderange> -d /xcat/iso/firmware/supermicro/p9/Boston_LC/20180205PNOR_BMC1.16/xCAT/
```

 @gurevichmark Could you run the verification on this one?

**Comment by gurevichmark** [comment]
Verified on `stratton01` Aug 22 build. 
Works as expected.
